cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -663,8 +671,16 @@ private ApplicationWithPrograms deployApp(NamespaceId namespaceId, @Nullable Str
                                             ArtifactDetail artifactDetail,
                                             @Nullable KerberosPrincipalId ownerPrincipal,
                                             boolean updateSchedules) throws Exception {
-    // Enforce that the current principal has write access to the namespace the app is being deployed to
-    authorizationEnforcer.enforce(namespaceId, authenticationContext.getPrincipal(), Action.WRITE);
+    KerberosPrincipalId principalId = ownerPrincipal == null ? null : ownerPrincipal;
```

Comment:
We cannot, since if it is null, that means no app impersonation is involved, and we need to pass null to later app deployment. If we make it as the namespace owner, then in later deployment, we will add it to the owner store, which will make this a app level impersonation